$OpenBSD: patch-mojoshader_opengl_c,v 1.3 2021/12/12 04:41:03 thfr Exp $

re-add MOJOSHADER_glProgramViewportFlip() for backwards compatibility.
Was removed by upstream in changeset 1210
(https://hg.icculus.org/icculus/mojoshader/rev/c586d4590241)

Index: mojoshader_opengl.c
--- mojoshader_opengl.c.orig
+++ mojoshader_opengl.c
@@ -2681,6 +2681,27 @@ static inline GLenum opengl_attr_type(const MOJOSHADER
     return GL_NONE;  // oh well. Raises a GL error later.
 } // opengl_attr_type
 
+#ifdef MOJOSHADER_FLIP_RENDERTARGET
+
+
+void MOJOSHADER_glProgramViewportFlip(int flip)
+{
+	assert(ctx->bound_program->vs_flip_loc != -1);
+
+	/* Some compilers require that vpFlip be a float value, rather than int.
+	 * However, there's no real reason for it to be a float in the API, so we
+	 * do a cast in here. That's not so bad, right...?
+	 * -flibit
+	 */
+	if (flip != ctx->bound_program->current_flip)
+	{
+		ctx->glUniform1f(ctx->bound_program->vs_flip_loc, (float) flip);
+		ctx->bound_program->current_flip = flip;
+	} // if
+}
+
+#endif
+
 
 int MOJOSHADER_glGetVertexAttribLocation(MOJOSHADER_usage usage, int index)
 {
